x-pack/plugins/security_solution/public/detections/components/value_lists_management_modal/modal.tsx

```diff
@@ -62,19 +66,23 @@ export const ValueListsModalComponent: React.FC<ValueListsModalProps> = ({
   );
 
   useEffect(() => {
-    if (deleteResult != null && deletingListIds.length > 0) {
-      setDeletingListIds([...deletingListIds.filter((id) => id !== deleteResult.id)]);
+    if (deleteResult != null) {
+      setDeletingListIds((ids) => [...ids.filter((id) => id !== deleteResult.id)]);
       fetchLists();
     }
-  }, [deleteResult, deletingListIds, fetchLists]);
+  }, [deleteResult, fetchLists]);
 
   const handleExport = useCallback(
-    async ({ ids }: { ids: string[] }) =>
-      exportList({ http, listId: ids[0], signal: new AbortController().signal }),
+    ({ id }: { id: string }) => {
+      setExportingListIds((ids) => [...ids, id]);
+      exportList({ http, listId: id, signal: new AbortController().signal })
+        .then((blob) => {
+          setExportDownload({ name: id, blob });
+        })
+        .finally(() => setExportingListIds((ids) => [...ids.filter((_id) => _id !== id)]));
```

Comment:
I take it we don't want to utilize the try, catch, finally blocks and just want to use promises?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch

There should be try, finally by its self we can use if we still prefer to use async/await. I am probably going to still use the more native async/await syntax rather than using promise based then's in my code fwiw. Seems more like what everyone else is still doing unless I'm missing something (which I could be).

Throughout other parts of the Kibana code base I see people using the try/finally so that's good news at least even though I haven't personally used it before.

Reply:
I don't have a preference between the two, this code just happened to shake out this way. However, I realize that we're no longer handling errors here so I need to fix that. BRB

Reply:
@FrankHassanabad I fixed this in https://github.com/elastic/kibana/pull/73532/commits/1a7f82b0b7ac988a6b0490f6e8c51a9f20adff3d